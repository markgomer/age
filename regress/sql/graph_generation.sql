/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */


LOAD 'age';

SET search_path = ag_catalog;

-- TESTS FOR COMPLETE GRAPH GENERATION

SELECT * FROM create_complete_graph('gp1',5,'edges','vertices');

SELECT COUNT(*) FROM gp1."edges";
SELECT COUNT(*) FROM gp1."vertices";

SELECT * FROM cypher('gp1', $$MATCH (a)-[e]->(b) RETURN e$$) as (n agtype);

SELECT * FROM create_complete_graph('gp1',5,'edges','vertices');

SELECT COUNT(*) FROM gp1."edges";
SELECT COUNT(*) FROM gp1."vertices";

SELECT * FROM create_complete_graph('gp2',5,'edges');

-- SHOULD FAIL
SELECT * FROM create_complete_graph('gp3',5, NULL);

SELECT * FROM create_complete_graph('gp4',NULL,NULL);

SELECT * FROM create_complete_graph(NULL,NULL,NULL);

-- Should error out because same labels are used for both vertices and edges
SELECT * FROM create_complete_graph('gp5',5,'label','label');

-- DROPPING GRAPHS
SELECT drop_graph('gp1', true);
SELECT drop_graph('gp2', true);


-- TESTS FOR BARBELL GRAPH GENERATION

SELECT * FROM age_create_barbell_graph('gp1',6,0,'vertices1',NULL,'edges1',NULL);
SELECT COUNT(*) FROM gp1."vertices1";
SELECT COUNT(*) FROM gp1."edges1";

SELECT * FROM age_create_barbell_graph('gp1',5,3,'vertices2','{}','edges2','{}');
SELECT COUNT(*) FROM gp1."vertices2";
SELECT COUNT(*) FROM gp1."edges2";

SELECT * FROM age_create_barbell_graph('gp1',4,1,'vertices3',NULL,'edges3',NULL);
SELECT COUNT(*) FROM gp1."vertices3";
SELECT COUNT(*) FROM gp1."edges3";

SELECT * FROM age_create_barbell_graph('gp1',3,2,'vertices4',NULL,'edges4',NULL);
SELECT COUNT(*) FROM gp1."vertices4";
SELECT COUNT(*) FROM gp1."edges4";

-- Counting total vertexes and edges
SELECT * FROM cypher('gp1', $$MATCH (v) RETURN COUNT(v)$$) as (n agtype);
SELECT * FROM cypher('gp1', $$MATCH (a)-[e]->(b) RETURN COUNT(e)$$) as (n agtype);

-- List all connections
SELECT * FROM cypher('gp1', $$MATCH (a)-[e]->(b) RETURN e$$) as (n agtype);

-- creating another graph
SELECT * FROM age_create_barbell_graph('gp2',10,10,'vertices',NULL,'edges',NULL);

-- testing vertex NULL labels
SELECT * FROM age_create_barbell_graph('gp3',6,6,NULL,NULL,'edges');

-- SHOULD FAIL
-- invalid graph name
SELECT * FROM age_create_barbell_graph(NULL,NULL,NULL,NULL,NULL,NULL,NULL);
SELECT * FROM age_create_barbell_graph(NULL,NULL,NULL);

-- invalid number of vertexes in complete graphs
SELECT * FROM age_create_barbell_graph('gp2',NULL,0,'vertices',NULL,'edges',NULL); 
SELECT * FROM age_create_barbell_graph('gp2',0,0,'vertices',NULL,'edges',NULL);
SELECT * FROM age_create_barbell_graph('gp2',1,1,'vertices',NULL,'edges',NULL);
SELECT * FROM age_create_barbell_graph('gp2',2,2,'vertices',NULL,'edges',NULL);
SELECT * FROM age_create_barbell_graph('gp2',-1,0,'vertices',NULL,'edges',NULL);

-- invalid number of vertexes in bridge
SELECT * FROM age_create_barbell_graph('gp2',3,-1,'vertices',NULL,'edges',NULL);
SELECT * FROM age_create_barbell_graph('gp3',5,NULL,'vertices',NULL,'edges',NULL);

-- invalid edge label
SELECT * FROM age_create_barbell_graph('gp5',5,3,'vertices',NULL,NULL,NULL);
SELECT * FROM age_create_barbell_graph('gp5',5,4,'vertices');

-- Should error out because same labels are used for both vertices and edges
SELECT * FROM age_create_barbell_graph('gp6',5,10,'label',NULL,'label',NULL);

-- DROPPING GRAPHS
SELECT drop_graph('gp1', true);
SELECT drop_graph('gp2', true);
SELECT drop_graph('gp3', true);


-- TESTS FOR CREATE PATH

--SELECT * FROM age_create_path('gp1',)