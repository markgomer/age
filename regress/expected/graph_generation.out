/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
LOAD 'age';
SET search_path = ag_catalog;
SELECT * FROM create_complete_graph('gp1',5,'edges','vertices');
NOTICE:  graph "gp1" has been created
NOTICE:  VLabel "vertices" has been created
NOTICE:  ELabel "edges" has been created
 create_complete_graph 
-----------------------
 
(1 row)

SELECT COUNT(*) FROM gp1."edges";
 count 
-------
    10
(1 row)

SELECT COUNT(*) FROM gp1."vertices";
 count 
-------
     5
(1 row)

SELECT * FROM cypher('gp1', $$MATCH (a)-[e]->(b) RETURN e$$) as (n agtype);
                                                             n                                                              
----------------------------------------------------------------------------------------------------------------------------
 {"id": 1125899906842625, "label": "edges", "end_id": 844424930131970, "start_id": 844424930131969, "properties": {}}::edge
 {"id": 1125899906842629, "label": "edges", "end_id": 844424930131971, "start_id": 844424930131970, "properties": {}}::edge
 {"id": 1125899906842626, "label": "edges", "end_id": 844424930131971, "start_id": 844424930131969, "properties": {}}::edge
 {"id": 1125899906842630, "label": "edges", "end_id": 844424930131972, "start_id": 844424930131970, "properties": {}}::edge
 {"id": 1125899906842627, "label": "edges", "end_id": 844424930131972, "start_id": 844424930131969, "properties": {}}::edge
 {"id": 1125899906842632, "label": "edges", "end_id": 844424930131972, "start_id": 844424930131971, "properties": {}}::edge
 {"id": 1125899906842631, "label": "edges", "end_id": 844424930131973, "start_id": 844424930131970, "properties": {}}::edge
 {"id": 1125899906842634, "label": "edges", "end_id": 844424930131973, "start_id": 844424930131972, "properties": {}}::edge
 {"id": 1125899906842633, "label": "edges", "end_id": 844424930131973, "start_id": 844424930131971, "properties": {}}::edge
 {"id": 1125899906842628, "label": "edges", "end_id": 844424930131973, "start_id": 844424930131969, "properties": {}}::edge
(10 rows)

SELECT * FROM create_complete_graph('gp1',5,'edges','vertices');
 create_complete_graph 
-----------------------
 
(1 row)

SELECT COUNT(*) FROM gp1."edges";
 count 
-------
    20
(1 row)

SELECT COUNT(*) FROM gp1."vertices";
 count 
-------
    10
(1 row)

SELECT * FROM create_complete_graph('gp2',5,'edges');
NOTICE:  graph "gp2" has been created
NOTICE:  ELabel "edges" has been created
 create_complete_graph 
-----------------------
 
(1 row)

-- SHOULD FAIL
SELECT * FROM create_complete_graph('gp3',5, NULL);
ERROR:  edge label can not be NULL
SELECT * FROM create_complete_graph('gp4',NULL,NULL);
ERROR:  number of nodes can not be NULL
SELECT * FROM create_complete_graph(NULL,NULL,NULL);
ERROR:  graph name can not be NULL
-- Should error out because same labels are used for both vertices and edges
SELECT * FROM create_complete_graph('gp5',5,'label','label');
ERROR:  vertex and edge label can not be same
-- DROPPING GRAPHS
SELECT drop_graph('gp1', true);
NOTICE:  drop cascades to 4 other objects
DETAIL:  drop cascades to table gp1._ag_label_vertex
drop cascades to table gp1._ag_label_edge
drop cascades to table gp1.vertices
drop cascades to table gp1.edges
NOTICE:  graph "gp1" has been dropped
 drop_graph 
------------
 
(1 row)

SELECT drop_graph('gp2', true);
NOTICE:  drop cascades to 3 other objects
DETAIL:  drop cascades to table gp2._ag_label_vertex
drop cascades to table gp2._ag_label_edge
drop cascades to table gp2.edges
NOTICE:  graph "gp2" has been dropped
 drop_graph 
------------
 
(1 row)

-- Tests for barbell graph generation
SELECT * FROM age_create_barbell_graph('gp1',5,0,'vertices',NULL,'edges',NULL);
NOTICE:  graph "gp1" has been created
NOTICE:  VLabel "vertices" has been created
NOTICE:  ELabel "edges" has been created
 age_create_barbell_graph 
--------------------------
 
(1 row)

SELECT COUNT(*) FROM gp1."edges";
 count 
-------
    21
(1 row)

SELECT COUNT(*) FROM gp1."vertices";
 count 
-------
    10
(1 row)

SELECT * FROM cypher('gp1', $$MATCH (a)-[e]->(b) RETURN e$$) as (n agtype);
                                                             n                                                              
----------------------------------------------------------------------------------------------------------------------------
 {"id": 1125899906842625, "label": "edges", "end_id": 844424930131970, "start_id": 844424930131969, "properties": {}}::edge
 {"id": 1125899906842629, "label": "edges", "end_id": 844424930131971, "start_id": 844424930131970, "properties": {}}::edge
 {"id": 1125899906842626, "label": "edges", "end_id": 844424930131971, "start_id": 844424930131969, "properties": {}}::edge
 {"id": 1125899906842627, "label": "edges", "end_id": 844424930131972, "start_id": 844424930131969, "properties": {}}::edge
 {"id": 1125899906842632, "label": "edges", "end_id": 844424930131972, "start_id": 844424930131971, "properties": {}}::edge
 {"id": 1125899906842630, "label": "edges", "end_id": 844424930131972, "start_id": 844424930131970, "properties": {}}::edge
 {"id": 1125899906842634, "label": "edges", "end_id": 844424930131973, "start_id": 844424930131972, "properties": {}}::edge
 {"id": 1125899906842628, "label": "edges", "end_id": 844424930131973, "start_id": 844424930131969, "properties": {}}::edge
 {"id": 1125899906842631, "label": "edges", "end_id": 844424930131973, "start_id": 844424930131970, "properties": {}}::edge
 {"id": 1125899906842633, "label": "edges", "end_id": 844424930131973, "start_id": 844424930131971, "properties": {}}::edge
 {"id": 1125899906842635, "label": "edges", "end_id": 844424930131975, "start_id": 844424930131974, "properties": {}}::edge
 {"id": 1125899906842639, "label": "edges", "end_id": 844424930131976, "start_id": 844424930131975, "properties": {}}::edge
 {"id": 1125899906842636, "label": "edges", "end_id": 844424930131976, "start_id": 844424930131974, "properties": {}}::edge
 {"id": 1125899906842637, "label": "edges", "end_id": 844424930131977, "start_id": 844424930131974, "properties": {}}::edge
 {"id": 1125899906842642, "label": "edges", "end_id": 844424930131977, "start_id": 844424930131976, "properties": {}}::edge
 {"id": 1125899906842640, "label": "edges", "end_id": 844424930131977, "start_id": 844424930131975, "properties": {}}::edge
 {"id": 1125899906842644, "label": "edges", "end_id": 844424930131978, "start_id": 844424930131977, "properties": {}}::edge
 {"id": 1125899906842638, "label": "edges", "end_id": 844424930131978, "start_id": 844424930131974, "properties": {}}::edge
 {"id": 1125899906842641, "label": "edges", "end_id": 844424930131978, "start_id": 844424930131975, "properties": {}}::edge
 {"id": 1125899906842643, "label": "edges", "end_id": 844424930131978, "start_id": 844424930131976, "properties": {}}::edge
 {"id": 1125899906842645, "label": "edges", "end_id": 844424930131978, "start_id": 844424930131969, "properties": {}}::edge
(21 rows)

SELECT * FROM age_create_barbell_graph('gp1',5,0,'vertices',NULL,'edges',NULL);
 age_create_barbell_graph 
--------------------------
 
(1 row)

SELECT COUNT(*) FROM gp1."edges";
 count 
-------
    42
(1 row)

SELECT COUNT(*) FROM gp1."vertices";
 count 
-------
    20
(1 row)

SELECT * FROM age_create_barbell_graph('gp2',5,10,'vertices',NULL,'edges',NULL);
NOTICE:  graph "gp2" has been created
NOTICE:  VLabel "vertices" has been created
NOTICE:  ELabel "edges" has been created
 age_create_barbell_graph 
--------------------------
 
(1 row)

-- SHOULD FAIL
SELECT * FROM age_create_barbell_graph(NULL,NULL,NULL,NULL,NULL,NULL,NULL);
ERROR:  Graph name cannot be NULL
SELECT * FROM age_create_barbell_graph('gp2',NULL,0,'vertices',NULL,'edges',NULL); 
ERROR:  Graph size cannot be NULL or lower than 3
SELECT * FROM age_create_barbell_graph('gp3',5,NULL,'vertices',NULL,'edges',NULL);
ERROR:  Bridge size cannot be NULL or lower than 0
SELECT * FROM age_create_barbell_graph('gp4',NULL,0,'vertices',NULL,'edges',NULL);
ERROR:  Graph size cannot be NULL or lower than 3
SELECT * FROM age_create_barbell_graph('gp5',5,0,'vertices',NULL,NULL,NULL);
ERROR:  edge label can not be NULL
-- Should error out because same labels are used for both vertices and edges
SELECT * FROM age_create_barbell_graph('gp6',5,10,'label',NULL,'label',NULL);
ERROR:  vertex and edge label can not be same
-- DROPPING GRAPHS
SELECT drop_graph('gp1', true);
NOTICE:  drop cascades to 4 other objects
DETAIL:  drop cascades to table gp1._ag_label_vertex
drop cascades to table gp1._ag_label_edge
drop cascades to table gp1.vertices
drop cascades to table gp1.edges
NOTICE:  graph "gp1" has been dropped
 drop_graph 
------------
 
(1 row)

SELECT drop_graph('gp2', true);
NOTICE:  drop cascades to 4 other objects
DETAIL:  drop cascades to table gp2._ag_label_vertex
drop cascades to table gp2._ag_label_edge
drop cascades to table gp2.vertices
drop cascades to table gp2.edges
NOTICE:  graph "gp2" has been dropped
 drop_graph 
------------
 
(1 row)


-- TESTS FOR CYCLE GRAPH GENERATION
SELECT * FROM create_cycle_graph('gp1',3,'edge1','{}','vertex1',NULL);
NOTICE:  graph "gp1" has been created
NOTICE:  VLabel "vertex1" has been created
NOTICE:  ELabel "edge1" has been created
 create_cycle_graph 
--------------------
 844424930131969
(1 row)

SELECT COUNT(*) FROM gp1."vertex1";
 count 
-------
     3
(1 row)

SELECT COUNT(*) FROM gp1."edge1";
 count 
-------
     3
(1 row)

SELECT * FROM create_cycle_graph('gp1',5,'edge2','{}','vertex2','{}');
NOTICE:  VLabel "vertex2" has been created
NOTICE:  ELabel "edge2" has been created
 create_cycle_graph 
--------------------
 1407374883553281
(1 row)

SELECT COUNT(*) FROM gp1."vertex2";
 count 
-------
     5
(1 row)

SELECT COUNT(*) FROM gp1."edge2";
 count 
-------
     5
(1 row)

SELECT * FROM create_cycle_graph('gp1',10,'edge3','{}','vertex3','{}');
NOTICE:  VLabel "vertex3" has been created
NOTICE:  ELabel "edge3" has been created
 create_cycle_graph 
--------------------
 1970324836974593
(1 row)

SELECT COUNT(*) FROM gp1."vertex3";
 count 
-------
    10
(1 row)

SELECT COUNT(*) FROM gp1."edge3";
 count 
-------
    10
(1 row)

SELECT * FROM create_cycle_graph('gp1',20,'edge4','{}','vertex4','{}');
NOTICE:  VLabel "vertex4" has been created
NOTICE:  ELabel "edge4" has been created
 create_cycle_graph 
--------------------
 2533274790395905
(1 row)

SELECT COUNT(*) FROM gp1."vertex4";
 count 
-------
    20
(1 row)

SELECT COUNT(*) FROM gp1."edge4";
 count 
-------
    20
(1 row)

-- Counting total vertexes and edges
SELECT * FROM cypher('gp1', $$MATCH (v) RETURN COUNT(v)$$) as (n agtype);
 n  
----
 38
(1 row)

SELECT * FROM cypher('gp1', $$MATCH (a)-[e]->(b) RETURN COUNT(e)$$) as (n agtype);
 n  
----
 38
(1 row)

-- List all connections
SELECT * FROM cypher('gp1', $$MATCH (a)-[e]->(b) RETURN e$$) as (n agtype);
                                                              n                                                               
------------------------------------------------------------------------------------------------------------------------------
 {"id": 1125899906842627, "label": "edge1", "end_id": 844424930131969, "start_id": 844424930131971, "properties": {}}::edge
 {"id": 1125899906842625, "label": "edge1", "end_id": 844424930131970, "start_id": 844424930131969, "properties": {}}::edge
 {"id": 1125899906842626, "label": "edge1", "end_id": 844424930131971, "start_id": 844424930131970, "properties": {}}::edge
 {"id": 1688849860263941, "label": "edge2", "end_id": 1407374883553281, "start_id": 1407374883553285, "properties": {}}::edge
 {"id": 1688849860263937, "label": "edge2", "end_id": 1407374883553282, "start_id": 1407374883553281, "properties": {}}::edge
 {"id": 1688849860263938, "label": "edge2", "end_id": 1407374883553283, "start_id": 1407374883553282, "properties": {}}::edge
 {"id": 1688849860263939, "label": "edge2", "end_id": 1407374883553284, "start_id": 1407374883553283, "properties": {}}::edge
 {"id": 1688849860263940, "label": "edge2", "end_id": 1407374883553285, "start_id": 1407374883553284, "properties": {}}::edge
 {"id": 2251799813685258, "label": "edge3", "end_id": 1970324836974593, "start_id": 1970324836974602, "properties": {}}::edge
 {"id": 2251799813685249, "label": "edge3", "end_id": 1970324836974594, "start_id": 1970324836974593, "properties": {}}::edge
 {"id": 2251799813685250, "label": "edge3", "end_id": 1970324836974595, "start_id": 1970324836974594, "properties": {}}::edge
 {"id": 2251799813685251, "label": "edge3", "end_id": 1970324836974596, "start_id": 1970324836974595, "properties": {}}::edge
 {"id": 2251799813685252, "label": "edge3", "end_id": 1970324836974597, "start_id": 1970324836974596, "properties": {}}::edge
 {"id": 2251799813685253, "label": "edge3", "end_id": 1970324836974598, "start_id": 1970324836974597, "properties": {}}::edge
 {"id": 2251799813685254, "label": "edge3", "end_id": 1970324836974599, "start_id": 1970324836974598, "properties": {}}::edge
 {"id": 2251799813685255, "label": "edge3", "end_id": 1970324836974600, "start_id": 1970324836974599, "properties": {}}::edge
 {"id": 2251799813685256, "label": "edge3", "end_id": 1970324836974601, "start_id": 1970324836974600, "properties": {}}::edge
 {"id": 2251799813685257, "label": "edge3", "end_id": 1970324836974602, "start_id": 1970324836974601, "properties": {}}::edge
 {"id": 2814749767106580, "label": "edge4", "end_id": 2533274790395905, "start_id": 2533274790395924, "properties": {}}::edge
 {"id": 2814749767106561, "label": "edge4", "end_id": 2533274790395906, "start_id": 2533274790395905, "properties": {}}::edge
 {"id": 2814749767106562, "label": "edge4", "end_id": 2533274790395907, "start_id": 2533274790395906, "properties": {}}::edge
 {"id": 2814749767106563, "label": "edge4", "end_id": 2533274790395908, "start_id": 2533274790395907, "properties": {}}::edge
 {"id": 2814749767106564, "label": "edge4", "end_id": 2533274790395909, "start_id": 2533274790395908, "properties": {}}::edge
 {"id": 2814749767106565, "label": "edge4", "end_id": 2533274790395910, "start_id": 2533274790395909, "properties": {}}::edge
 {"id": 2814749767106566, "label": "edge4", "end_id": 2533274790395911, "start_id": 2533274790395910, "properties": {}}::edge
 {"id": 2814749767106567, "label": "edge4", "end_id": 2533274790395912, "start_id": 2533274790395911, "properties": {}}::edge
 {"id": 2814749767106568, "label": "edge4", "end_id": 2533274790395913, "start_id": 2533274790395912, "properties": {}}::edge
 {"id": 2814749767106569, "label": "edge4", "end_id": 2533274790395914, "start_id": 2533274790395913, "properties": {}}::edge
 {"id": 2814749767106570, "label": "edge4", "end_id": 2533274790395915, "start_id": 2533274790395914, "properties": {}}::edge
 {"id": 2814749767106571, "label": "edge4", "end_id": 2533274790395916, "start_id": 2533274790395915, "properties": {}}::edge
 {"id": 2814749767106572, "label": "edge4", "end_id": 2533274790395917, "start_id": 2533274790395916, "properties": {}}::edge
 {"id": 2814749767106573, "label": "edge4", "end_id": 2533274790395918, "start_id": 2533274790395917, "properties": {}}::edge
 {"id": 2814749767106574, "label": "edge4", "end_id": 2533274790395919, "start_id": 2533274790395918, "properties": {}}::edge
 {"id": 2814749767106575, "label": "edge4", "end_id": 2533274790395920, "start_id": 2533274790395919, "properties": {}}::edge
 {"id": 2814749767106576, "label": "edge4", "end_id": 2533274790395921, "start_id": 2533274790395920, "properties": {}}::edge
 {"id": 2814749767106577, "label": "edge4", "end_id": 2533274790395922, "start_id": 2533274790395921, "properties": {}}::edge
 {"id": 2814749767106578, "label": "edge4", "end_id": 2533274790395923, "start_id": 2533274790395922, "properties": {}}::edge
 {"id": 2814749767106579, "label": "edge4", "end_id": 2533274790395924, "start_id": 2533274790395923, "properties": {}}::edge
(38 rows)

-- creating another graph
SELECT * FROM create_cycle_graph('gp2',7,'edge1','{}','vertex1','{}');
NOTICE:  graph "gp2" has been created
NOTICE:  VLabel "vertex1" has been created
NOTICE:  ELabel "edge1" has been created
 create_cycle_graph 
--------------------
 844424930131969
(1 row)

-- testing vertex NULL labels
SELECT * FROM create_cycle_graph('gp3',6,'edge','{}',NULL,NULL);
NOTICE:  graph "gp3" has been created
NOTICE:  ELabel "edge" has been created
 create_cycle_graph 
--------------------
 281474976710657
(1 row)

-- SHOULD FAIL
-- invalid graph name
SELECT * FROM create_cycle_graph(NULL,6,'edge','{}','node','{}');
ERROR:  graph name cannot be NULL
SELECT * FROM create_cycle_graph(NULL,6,'edge');
ERROR:  graph name cannot be NULL
-- invalid number of vertexes in cycle graphs
SELECT * FROM create_cycle_graph('ERROR1',0,'edge',NULL,'node',NULL); 
ERROR:  cycle graph size cannot be NULL or lower than 3
SELECT * FROM create_cycle_graph('ERROR1',0,'edge',NULL,'node',NULL);
ERROR:  cycle graph size cannot be NULL or lower than 3
SELECT * FROM create_cycle_graph('ERROR1',1,'edge',NULL,'node',NULL);
ERROR:  cycle graph size cannot be NULL or lower than 3
SELECT * FROM create_cycle_graph('ERROR1',2,'edge',NULL,'node',NULL);
ERROR:  cycle graph size cannot be NULL or lower than 3
SELECT * FROM create_cycle_graph('ERROR1',-1,'edge',NULL,'node',NULL);
ERROR:  cycle graph size cannot be NULL or lower than 3
-- invalid edge label
SELECT * FROM create_cycle_graph('ERROR2',5,NULL,'{}','vertex2','{}');
ERROR:  edge label cannot be NULL
SELECT * FROM create_cycle_graph('ERROR2',5,NULL);
ERROR:  edge label cannot be NULL
-- same edge and node labels
SELECT * FROM create_cycle_graph('ERROR',5,'same','{}','same','{}');
ERROR:  Vertex and edge label cannot be the same
-- DROPPING GRAPHS
SELECT drop_graph('gp1', true);
NOTICE:  drop cascades to 10 other objects
DETAIL:  drop cascades to table gp1._ag_label_vertex
drop cascades to table gp1._ag_label_edge
drop cascades to table gp1.vertex1
drop cascades to table gp1.edge1
drop cascades to table gp1.vertex2
drop cascades to table gp1.edge2
drop cascades to table gp1.vertex3
drop cascades to table gp1.edge3
drop cascades to table gp1.vertex4
drop cascades to table gp1.edge4
NOTICE:  graph "gp1" has been dropped
 drop_graph 
------------
 
(1 row)

SELECT drop_graph('gp2', true);
NOTICE:  drop cascades to 4 other objects
DETAIL:  drop cascades to table gp2._ag_label_vertex
drop cascades to table gp2._ag_label_edge
drop cascades to table gp2.vertex1
drop cascades to table gp2.edge1
NOTICE:  graph "gp2" has been dropped
 drop_graph 
------------
 
(1 row)

SELECT drop_graph('gp3', true);
NOTICE:  drop cascades to 3 other objects
DETAIL:  drop cascades to table gp3._ag_label_vertex
drop cascades to table gp3._ag_label_edge
drop cascades to table gp3.edge
NOTICE:  graph "gp3" has been dropped
 drop_graph 
------------
 
(1 row)
